src/Repositories/RedirectRepository.php

```diff
@@ -0,0 +1,390 @@
+<?php
+
+namespace Bonnier\WP\Redirect\Repositories;
+
+use Bonnier\WP\Redirect\Database\DB;
+use Bonnier\WP\Redirect\Database\Exceptions\DuplicateEntryException;
+use Bonnier\WP\Redirect\Database\Migrations\Migrate;
+use Bonnier\WP\Redirect\Database\Query;
+use Bonnier\WP\Redirect\Exceptions\IdenticalFromToException;
+use Bonnier\WP\Redirect\Helpers\LocaleHelper;
+use Bonnier\WP\Redirect\Helpers\UrlHelper;
+use Bonnier\WP\Redirect\Models\Redirect;
+use Illuminate\Support\Collection;
+
+class RedirectRepository extends BaseRepository
+{
+    /**
+     * RedirectRepository constructor.
+     * @param DB $database
+     * @throws \Exception
+     */
+    public function __construct(DB $database)
+    {
+        $this->tableName = Migrate::REDIRECTS_TABLE;
+        parent::__construct($database);
+    }
+
+    /**
+     * @param int $redirectID
+     * @return Redirect|null
+     */
+    public function getRedirectById(int $redirectID): ?Redirect
+    {
+        $data = $this->database->findById($redirectID);
+        if ($data) {
+            $redirect = new Redirect();
+            return $redirect->fromArray($data);
+        }
+
+        return null;
+    }
+
+    /**
+     * @param string $path
+     * @param string|null $locale
+     * @return Redirect|null
+     * @throws \Exception
+     */
+    public function findRedirectByPath(string $path, string $locale = null): ?Redirect
+    {
+        if ($redirect = $this->findExactRedirectByPath($path, $locale)) {
```

Comment:
```suggestion
       $redirect = $this->findExactRedirectByPath($path, $locale)) ?:
       $this->findRedirectByIgnoringQueryParams($path, $locale) ?:
       $this->findWildcardRedirectsByPath($path, $locale);
       
       return $redirect ? $this->handleQueryParams($redirect, $path) : null;
```

Reply:
I see where you're going, but I prefer the "unwrapped" version, because it's more clear to me what is actually happening, than using a smart ternary operator.